### ISSUE TITLE
The metrics and stats about objects and buckets not right 

### ISSUE THREAD

**Issue by HaykManukyanAvetiky** [minio/minio#12990]
I have an minio cluster with 12 nods and trying to load ~6m objects in it.
my objects are structured following way
main
    - part1(~3m objs)
          -- 2017-01-01(~1k objs, pics)
          -- ......
    - part2(~3m objs)
           --same as part1
and I have created bucket with following simple script 
```bash
#!/usr/bin/env bash

# USAGE
# ./create_bucket.sh <<cluster alias>> <<bucket name>>

set -e

mc mb --insecure $1/$2
echo "Bucket created"
mc version enable $1/$2
echo "Versioning enabled"
mc ilm import $1/$2 < version_ilm.json
echo "Lifecycle managment set up"
mc policy set download $1/$2
echo "Download ploicy attached"

echo "$1 bucket in $2 cluster is created successfuly" 
echo "with versioning, download policy and ilm specified in utils/version_ilm.json"
```
and the version_ilm.json is the following
```json
{
    "Rules": [
        {
            "ID": "Removing all old versions",
            "Filter": {
                "Prefix": "*"
            },
            "NoncurrentVersionExpiration": {
                "NoncurrentDays": 30
            },
            "NoncurrentVersionTransition": {
                "NoncurrentDays": 3,
                "StorageClass": "REDUCED_REDUNDANCY"
            },
            "Status": "Enabled"
        }
    ]
}
```

also I am using rclone to sync my data 

after aprox 500k objects I am getting wired spices and cut offs in number of objects and actual data 
![image](https://user-images.githubusercontent.com/65945812/129739836-44cd2efa-8437-43d5-bedc-3b4fea0bd2fa.png)
The point is I don't see any delete activity pic or so, it seems ok 
![image](https://user-images.githubusercontent.com/65945812/129740326-958b3255-433b-4de8-81a6-5dde1d5c25eb.png)

## Current Behavior
after this cut off grafana and console shows me  about 200k objects bat in minio server storage I can see much more.
also metrics and console not showing correct data about buckets, particularly not showing new buckets and still showing old ones.
Also mc ls unable to ls dirs with mire that 200k objects
![image](https://user-images.githubusercontent.com/65945812/129743040-079567df-0239-4bea-979f-3605af2e7824.png)
it throws this wired error after ~15m running, but if you see above it lists smaller bucket

**Comment by klauspost** [comment]
@HaykManukyanAvetiky You fail to mention which version you are running. 

If you are not running the latest be sure to upgrade. A fix to accounting was recently merged.

**Comment by harshavardhana** [comment]
> Also mc ls unable to ls dirs with mire that 200k objects

This is because your `mc` and server time are too large i.e make sure to sync then with `NTP`

**Comment by harshavardhana** [comment]
> ```json
>             "NoncurrentVersionTransition": {
>                 "NoncurrentDays": 3,
>                 "StorageClass": "REDUCED_REDUNDANCY"
>             },
> ```

This doesn't make sense

**Comment by HaykManukyanAvetiky** [comment]
> @HaykManukyanAvetiky You fail to mention which version you are running.
> 
> If you are not running the latest be sure to upgrade. A fix to accounting was recently merged.
I am running RELEASE.2021-08-05T22-01-19Z I will update to RELEASE.2021-08-17T20-53-08Z and keep you updated

**Comment by HaykManukyanAvetiky** [comment]
> ```json
> NoncurrentVersionTransition
> ```

@harshavardhana thanks for info I will remove that, I have used this docs [1](https://docs.aws.amazon.com/AmazonS3/latest/API/API_NoncurrentVersionTransition.html) [2](https://docs.aws.amazon.com/AmazonS3/latest/userguide/intro-lifecycle-rules.html) for S3 but seems that not right.

**Comment by HaykManukyanAvetiky** [comment]
> > Also mc ls unable to ls dirs with mire that 200k objects
> 
> This is because your `mc` and server time are too large i.e make sure to sync then with `NTP`

@harshavardhana I am not sure in that cause I list same cluster but different bucket right before failed  and it worked, you can see that in the picture. ```mc ls main/thumbnails``` is working but ```mc ls main/new-hound/thumbnails``` not. both buckets are in same cluster ```main```.  The difference in this buckets are number of objects in `new-hound/thumbnails` much more than in  `thumbnails`, also thumbnails dont have any lifecycle rule.

**Comment by harshavardhana** [comment]
> @harshavardhana I am not sure in that cause I list same cluster but different bucket right before failed and it worked, you can see that in the picture. `mc ls main/thumbnails` is working but `mc ls main/new-hound/thumbnails` not. both buckets are in same cluster `main`. The difference in this buckets are number of objects in `new-hound/thumbnails` much more than in `thumbnails`, also thumbnails dont have any lifecycle rule.

You have versioning enabled make sure to list using `--versions` @HaykManukyanAvetiky

**Comment by HaykManukyanAvetiky** [comment]
@harshavardhana  thanks for response, seems I fount the reason why I lose lots of objects, it comes from rclone side. the cases with metrics still strange, I cant find much info about them and I think there is some latency. I have changed the original grafana dashboard, I think it much better now, I will be happy to share it with you. Also I would like to do some feature requests for mc, particularly 
- PUT or DELETE tag for mc undo to revert specific action, would be really handy for data recovery, now it reverts all object and it can be reason for data lose if not specify object name precisely.
- mc ls have option revers of --rewind to get versions  lather than

I tried to open feature request for mc but there only bag option in repo.
And a question I tried to understand when and how minio applies ilms ? 
Is there some scheduled task or so ? is there some logging fore that ?
can i manually trigger ilm ?
